java/flight/pom.xml

```diff
@@ -95,6 +95,11 @@
       <artifactId>protobuf-java</artifactId>
       <version>3.5.1</version>
     </dependency>
+    <dependency>
```

Comment:
why is this necessary if we already import grpc?

Reply:
So it seems like we import specific "sub-packages" and when I bumped the version up, it required this package. I think the cleaner way to do it would to use grpc-all (https://mvnrepository.com/artifact/io.grpc/grpc-all)

Reply:
Let's avoid doing all unless we need to

Reply:
This is the error while compiling if we remove the dependency:

[WARNING] Used undeclared dependencies found:
[WARNING]    io.grpc:grpc-api:jar:1.24.0:compile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  35.102 s
[INFO] Finished at: 2019-10-08T12:46:08-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.0.1:analyze-only (analyze) on project arrow-flight: Dependency problems found -> [Help 1]

Reply:
api is the right import, let's keep this as was done. Someone added a dependency check step that ensures: If we refer to something specifically, we need to include it as a dependency, even if it is already included as a transitive dependency (to answer @emkornfield's question).